osbenchmark/workload/params.py

```diff
@@ -1140,17 +1189,34 @@ def params(self):
         return self.query_params
 
     def _build_vector_search_query_body(self, vector, efficient_filter=None) -> dict:
-        """Builds a k-NN request that can be used to execute an approximate nearest
+        """Builds a vector search request that can be used to execute an approximate nearest
         neighbor search against a k-NN plugin index
         Args:
             vector: vector used for query
+            efficient_filter: efficient filter used for query
         Returns:
             A dictionary containing the body used for search query
         """
-        query = {
+        query = {}
+        if self.query_type == self.KNN_QUERY_TYPE:
+            query.update({
+                "k": self.k,
+            })
```

Comment:
Instead of using `query.update()` here:
```
query_update["k"] = self.k
```

Reply:
I'm following the same pattern used in this function.

Reply:
Again, this is for better readability, especially if there is only a single key being updated.  I won't insist, though.